<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>BackendSettingsPage</class>
 <widget class="QWidget" name="BackendSettingsPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>596</width>
    <height>817</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Backend</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <item>
    <widget class="QGroupBox" name="groupbox_audio_output">
     <property name="title">
      <string>Audio output</string>
     </property>
     <layout class="QVBoxLayout" name="layout_audio_output">
      <item>
       <layout class="QHBoxLayout" name="layout_engine">
        <property name="spacing">
         <number>10</number>
        </property>
        <item>
         <widget class="QLabel" name="label_engine">
          <property name="text">
           <string>Engine</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="combobox_engine">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="minimumSize">
           <size>
            <width>400</width>
            <height>0</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="spacer_engine">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="layout_output">
        <property name="spacing">
         <number>10</number>
        </property>
        <item>
         <widget class="QLabel" name="label_output">
          <property name="text">
           <string>Output</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="combobox_output">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="minimumSize">
           <size>
            <width>400</width>
            <height>0</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="spacer_output">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="layout_device">
        <property name="spacing">
         <number>10</number>
        </property>
        <item>
         <widget class="QLabel" name="label_device">
          <property name="text">
           <string>Device</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="combobox_device">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="minimumSize">
           <size>
            <width>240</width>
            <height>0</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="LineEdit" name="lineedit_device" native="true">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="minimumSize">
           <size>
            <width>90</width>
            <height>30</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>160</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="hint" stdset="0">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="spacer_device">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QWidget" name="widget_alsa_plugin" native="true">
        <layout class="QHBoxLayout" name="layout_alsa_plugin">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="label_alsaplugin">
           <property name="text">
            <string>ALSA plugin</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radiobutton_alsa_hw">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>hw</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radiobutton_alsa_plughw">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>p&amp;lughw</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="spacer_alsaplugin">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="checkbox_volume_control">
        <property name="text">
         <string>Enable volume control</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupbox_buffer">
     <property name="title">
      <string>Buffer</string>
     </property>
     <layout class="QFormLayout" name="layout_buffer">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
      </property>
      <item row="1" column="0">
       <widget class="QLabel" name="label_bufferduration">
        <property name="text">
         <string>Buffer duration</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="spinbox_bufferduration">
        <property name="suffix">
         <string> ms</string>
        </property>
        <property name="maximum">
         <number>60000</number>
        </property>
        <property name="singleStep">
         <number>100</number>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_bufferminfill">
        <property name="text">
         <string>Minimum buffer fill</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_buffer">
        <item>
         <widget class="QLabel" name="label_bufferminfillvalue">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="slider_bufferminfill">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>50</number>
          </property>
          <property name="value">
           <number>33</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="tickInterval">
           <number>1</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupbox_replaygain">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="title">
      <string>Replay Gain</string>
     </property>
     <layout class="QVBoxLayout" name="layout_replay_gain">
      <item>
       <widget class="QCheckBox" name="checkbox_replaygain">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>Use Replay Gain metadata if it is available</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="widget_replaygain" native="true">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <layout class="QFormLayout" name="formLayout_4">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="label_regainmode">
           <property name="text">
            <string>Replay Gain mode</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="combobox_replaygainmode">
           <item>
            <property name="text">
             <string>Radio (equal loudness for all tracks)</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Album (ideal loudness for all tracks)</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_preamp">
           <property name="text">
            <string>Pre-amp</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="QLabel" name="label_replaygainpreamp"/>
           </item>
           <item>
            <widget class="StickySlider" name="stickslider_replaygainpreamp">
             <property name="maximum">
              <number>300</number>
             </property>
             <property name="value">
              <number>150</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sticky_center" stdset="0">
              <number>150</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="2" column="0" colspan="2">
          <widget class="QCheckBox" name="checkbox_replaygaincompression">
           <property name="text">
            <string>Apply compression to prevent clipping</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupbox_fading">
     <property name="title">
      <string>Fading</string>
     </property>
     <layout class="QVBoxLayout" name="layout_fading">
      <item>
       <widget class="QCheckBox" name="checkbox_fadeout_stop">
        <property name="text">
         <string>Fade out when stopping a track</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="checkbox_fadeout_cross">
        <property name="text">
         <string>Cross-fade when changing tracks manually</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="checkbox_fadeout_auto">
        <property name="text">
         <string>Cross-fade when changing tracks automatically</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="checkbox_fadeout_samealbum">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Except between tracks on the same album or in the same CUE sheet</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="widget_fading_options" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="label_fadingduration_1">
           <property name="text">
            <string>Fading duration</string>
           </property>
           <property name="indent">
            <number>22</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinbox_fadeduration">
           <property name="suffix">
            <string> ms</string>
           </property>
           <property name="maximum">
            <number>10000</number>
           </property>
           <property name="singleStep">
            <number>1000</number>
           </property>
           <property name="value">
            <number>2000</number>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="spacer_fading_1">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>257</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="checkbox_fadeout_pauseresume">
        <property name="text">
         <string>Fade out on pause / fade in on resume</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="layout_fading_duration">
        <item>
         <widget class="QLabel" name="label_fading_duration_1">
          <property name="text">
           <string>Fading duration</string>
          </property>
          <property name="indent">
           <number>22</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spinbox_fadeduration_pauseresume">
          <property name="suffix">
           <string> ms</string>
          </property>
          <property name="maximum">
           <number>10000</number>
          </property>
          <property name="singleStep">
           <number>50</number>
          </property>
          <property name="value">
           <number>250</number>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="spacer_fading_duration_1">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="spacer_bottom">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>StickySlider</class>
   <extends>QSlider</extends>
   <header>widgets/stickyslider.h</header>
  </customwidget>
  <customwidget>
   <class>LineEdit</class>
   <extends>QWidget</extends>
   <header>widgets/lineedit.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>checkbox_replaygain</sender>
   <signal>toggled(bool)</signal>
   <receiver>widget_replaygain</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>89</x>
     <y>259</y>
    </hint>
    <hint type="destinationlabel">
     <x>143</x>
     <y>285</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
